src/main/java/koreatech/in/dto/normal/shop/response/AllMenusOfShopResponse.java

```diff
@@ -61,14 +74,36 @@ public static class OptionPrice {
         }
     }
 
-    public static AllMenusOfShopResponse from(List<ShopMenuProfile> shopMenuProfiles) {
+    public static AllMenusOfShopResponse from(List<ShopMenuProfile> shopMenuProfiles, List<ShopCategory> categoryNames) {
+        List<Category> category = new ArrayList<>();
+        List<Integer> categoryIds = shopMenuProfiles.stream()
+                .map(ShopMenuProfile::getCategory_ids)
+                .flatMap(Collection::stream)
+                .distinct()
+                .sorted()
+                .collect(Collectors.toList());
+        for (int i = 0; i < categoryIds.size(); i++) {
+            int categoryIndex = categoryIds.get(i);
+            category.add(
+                    Category.builder()
+                            .id(categoryIndex)
+                            .name(
+                                    categoryNames.stream()
+                                            .filter(categoryName -> categoryName.getId().equals(categoryIndex))
+                                            .collect(Collectors.toList()).get(0).getName()
```

Comment:
`get(0)` 이 존재한다는 근거가 빈약합니다. 
DB상에 NOT_NULL이라고 해서 무조건 존재한다고 가정하는 건 위험할 것 같습니다. 

-> 리스트로 만들고 0번째 원소를 가져올 거면 다음과 같이 구성해야 할 것 같습니다.
```suggestion
categoryNames.stream()
.filter(categoryName -> categoryName.getId().equals(categoryIndex))
.findFirst()
.orElse({예외 처리 로직})
.getName()
```

- 일반적으로는 DB 상에서 join을 이용해 shopId와 함께 shopName을 가져오는 방식을 취할것 같습니다. (게시판 서비스에서 댓글 사용자의 ID와 이름 모두 가져오는 것처럼)

Reply:
확실한 근거가 없는 가정은 위험할 수 있다!
기억해두겠습니다..!!